src/modules/workspaces/workspaces.controller.spec.ts

```diff
@@ -154,6 +154,22 @@ describe('Workspace Controller', () => {
     });
   });
 
+  describe('PATCH /:workspaceId', () => {
+    it('When workspace details are updated successfully, then it should return.', async () => {
+      const user = newUser();
+      const workspace = newWorkspace({ owner: user });
+
+      workspacesController.editWorkspaceDetails(workspace.id, user, {
+        name: 'new name',
+      });
+
+      expect(workspacesUsecases.editWorkspaceDetails).toHaveBeenCalledWith(
```

Comment:
This test is not asserting that the operation has finished successfully unless the method is synchronous, which is not the case. You should do something like: 

`await expect(workspacesController.editWorkspaceDetails(workspace.id, user, { name: 'new name' }).resolves();` or matching the returned content with the injected one by your use case being stubbed.

_And previous to that_, spy the use case to check that, in fact, is has been called with the proper parameters, as a way of improving the test.